crates/rattler_solve/src/resolvo/conda_sorting.rs

```diff
@@ -0,0 +1,426 @@
+            Ordering::Less => return Ordering::Less,
+            Ordering::Greater => return Ordering::Greater,
+            Ordering::Equal => {}
+        };
+
+        // Otherwise, select the variant with the highest version
+        match (self.strategy, a_record.version().cmp(b_record.version())) {
+            (CompareStrategy::Default, Ordering::Greater)
+            | (CompareStrategy::LowestVersion, Ordering::Less) => return Ordering::Less,
+            (CompareStrategy::Default, Ordering::Less)
+            | (CompareStrategy::LowestVersion, Ordering::Greater) => return Ordering::Greater,
+            (_, Ordering::Equal) => {}
+        };
+
+        // Otherwise, select the variant with the highest build number
+        match a_record.build_number().cmp(&b_record.build_number()) {
+            Ordering::Less => Ordering::Greater,
+            Ordering::Greater => Ordering::Less,
+            Ordering::Equal => Ordering::Equal,
+        }
+    }
+
+    fn sort_by_highest_dependency_versions(
+        &self,
+        solvables: &mut [SolvableId],
+        version_cache: &mut HashMap<VersionSetId, Option<(Version, bool)>>,
+    ) {
+        // Because the list can contain multiple versions, tracked features, and builds of the same package
+        // we need to create sub list of solvables that have the same version, build, and tracked features
+        // and sort these sub lists by the highest version of the dependencies shared by the solvables.
+        let mut start = 0usize;
+        let entire_len = solvables.len();
+        while start < entire_len {
+            let mut end = start + 1;
+
+            // Find the range of solvables with the same: version, build, tracked features
+            while end < entire_len
+                && self.simple_compare(solvables[start], solvables[end]) == Ordering::Equal
+            {
+                end += 1;
+            }
+
+            // Take the sub list of solvables
+            let sub = &mut solvables[start..end];
+            if sub.len() > 1 {
+                // Sort the sub list of solvables by the highest version of the dependencies
+                self._sort_by_highest_dependency_versions(sub, version_cache);
+            }
+
+            start = end;
+        }
+    }
+
+    /// Sorts the solvables by the highest version of the dependencies shared by the solvables.
+    /// what this function does is:
+    /// 1. Find the first unsorted solvable in the list
+    /// 2. Get the dependencies for each solvable
+    /// 3. Get the known dependencies for each solvable, filter out the unknown dependencies
+    /// 4. Retain the dependencies that are shared by all the solvables
+    /// 6. Calculate a total score by counting the position of the solvable in the list with sorted dependencies
+    /// 7. Sort by the score per solvable and use timestamp of the record as a tie breaker
+    fn _sort_by_highest_dependency_versions(
```

Comment:
The underscore would suggest its unused. Maybe rename to sort_subset_..

Reply:
Oops got some python confusion there.